localstack/services/stepfunctions/asl/component/state/state_execution/state_map/item_reader/reader_config/max_items_decl.py

```diff
@@ -48,7 +65,45 @@ class MaxItemsPath(MaxItemsDecl):
     def __init__(self, path: str):
         self.path: Final[str] = path
 
+    def _validate_value(self, env: Environment, value: int) -> None:
+        if not isinstance(value, int):
+            # TODO: Note, this error appears to be validated at a earlier stage in AWS Step Functions, unlike the
+            #  negative integer check that is validated at this exact depth.
+            error_typ = StatesErrorNameType.StatesItemReaderFailed
+            raise FailureEventException(
+                failure_event=FailureEvent(
+                    env=env,
+                    error_name=StatesErrorName(typ=error_typ),
+                    event_type=HistoryEventType.ExecutionFailed,
+                    event_details=EventDetails(
+                        executionFailedEventDetails=ExecutionFailedEventDetails(
+                            error=error_typ.to_name(),
+                            cause=(
+                                f"The MaxItemsPath field refers to value '{value}' "
+                                f"which is not a valid integer: {self.path}"
+                            ),
+                        )
+                    ),
+                )
+            )
+        if value < 0:
+            error_typ = StatesErrorNameType.StatesItemReaderFailed
+            raise FailureEventException(
+                failure_event=FailureEvent(
+                    env=env,
+                    error_name=StatesErrorName(typ=error_typ),
+                    event_type=HistoryEventType.MapRunFailed,
+                    event_details=EventDetails(
+                        executionFailedEventDetails=ExecutionFailedEventDetails(
+                            error=error_typ.to_name(),
+                            cause="field MaxItems must be positive",
+                        )
+                    ),
+                )
+            )
+
     def _get_value(self, env: Environment) -> int:
-        input_expr = parse(self.path)
-        max_items = input_expr.find(env.inp)
+        inp = env.stack[-1]
+        max_items = JSONPathUtils.extract_json(self.path, inp)
```

Comment:
good to fix more occurrences of JSON extraction logic following https://github.com/localstack/localstack/pull/10521 👍